### ISSUE TITLE
Handle NaN as input to Bar()

### ISSUE THREAD

**Issue by birdsarah** [bokeh/bokeh#2288]
From the mailing list: https://groups.google.com/a/continuum.io/forum/?utm_medium=email&utm_source=footer#!msg/bokeh/MsAVI0m-eyo/iNOBA9kTkNQJ

Should gracefully handle NaN in the data. 

Probably want to wait till after charts interface settles and handle NaN in similar way to TimeSeries, as mentioned in #1373.

**Comment by VelizarVESSELINOV** [comment]
Same for Area charts

Take any example and replace one of the values with nan and the results it is:
```
ValueError: Out of range float values are not JSON compliant: nan
```

**Comment by bryevdv** [comment]
This should be resolved by #5544

**Comment by bryevdv** [comment]
It was:

```
from bokeh.charts import Bar, output_file, show
from bokeh.charts.attributes import cat, color
from bokeh.charts.operations import blend
from bokeh.charts.utils import df_from_json
from bokeh.sampledata.olympics2014 import data

# utilize utility to make it easy to get json/dict data converted to a dataframe
df = df_from_json(data)

# filter by countries with at least one medal and sort by total medals
df = df[df['total'] > 0]
df = df.sort_values(by="total", ascending=False)

import numpy
df['bronze'][65] = numpy.nan

bar = Bar(df,
          values=blend('bronze', 'silver', 'gold', name='medals', labels_name='medal'),
          label=cat(columns='abbr', sort=False),
          stack=cat(columns='medal', sort=False),
          color=color(columns='medal', palette=['SaddleBrown', 'Silver', 'Goldenrod'],
                      sort=False),
          legend='top_right',
          title="Medals per Country, Sorted by Total Medals",
          tooltips=[('medal', '@medal'), ('country', '@abbr')])


output_file("stacked_bar.html", title="stacked_bar.py example")

show(bar)
```
generates:

<img width="664" alt="screen shot 2017-03-25 at 13 05 09" src="https://cloud.githubusercontent.com/assets/1078448/24324842/b17f6762-115b-11e7-8fc0-bac8b7a9e506.png">
